### ISSUE TITLE
Database Closed Exception when inserting data after resetting a database

### ISSUE THREAD

**Issue by danvwells** [Raizlabs/DBFlow#1115]
DBFlow Version: 4.0.0-beta3
Issue Kind (Bug, Question, Feature): Bug

Description: 
In our app, I wanted to wipe out all data in the database and then start again right away without having to add each Model class to a Delete (so that as we add models, we don't have to remember to add them there). When trying to reset the database and then use it again, I got errors saying the database was closed. I was able to reproduce this with a unit test in the latest DBFlow source on develop.

**Comment by danvwells** [comment]
see pull request: https://github.com/Raizlabs/DBFlow/pull/1112

**Comment by danvwells** [comment]
I noticed #925 and #1089 are very similar... could be viewed as a duplicate, however my issue happened when using a different SQLiteOpenHelper than FlowSQLiteOpenHelper so we can use a targeted version of SQLite that is not SQLCipher or the stock Android version.  The unit test in pull request #1112 broke even with the fix from #925, but the changes in pull request #1112 fixes it.

**Comment by agrosner** [comment]
you will have to manually add it yourself then. Unfortunately. 
Something similar:
```java
@Override
    public DatabaseWrapper getDatabase() {
        if (androidDatabase == null || !androidDatabase.getDatabase().isOpen()) {
            androidDatabase = AndroidDatabase.from(getWritableDatabase());
        }
        return androidDatabase;
    }

```
